From a575810206dfed719cc32bc0a0915f57faf7068d Mon Sep 17 00:00:00 2001
From: linjh <linjh@rock-chips.com>
Date: Thu, 27 Sep 2012 17:51:20 +0800
Subject: [PATCH] phonepad: update phonepad tp struct, exp gpio struct and lcd
 io enable control

[reference file]

	modified:
		arch/arm/mach-rk2928/include/mach/board.h
		include/linux/rk_screen.h
---
 arch/arm/mach-rk2928/include/mach/board.h | 20 ++++++++++++++++++++
 include/linux/rk_screen.h                 |  2 ++
 2 files changed, 22 insertions(+)

diff --git a/arch/arm/mach-rk2928/include/mach/board.h b/arch/arm/mach-rk2928/include/mach/board.h
index f6d6e99b2bdd..9cda9b8badd8 100644
--- a/arch/arm/mach-rk2928/include/mach/board.h
+++ b/arch/arm/mach-rk2928/include/mach/board.h
@@ -67,6 +67,26 @@ struct ft5x0x_platform_data{
     void    (*exit_platform_hw)(void);
 };
 #endif
+
+#if defined (CONFIG_TOUCHSCREEN_I30)
+struct ft5306_platform_data {
+    int rest_pin;
+    int irq_pin ;
+    int     (*get_pendown_state)(void);
+    int     (*init_platform_hw)(void);
+    int     (*platform_sleep)(void);
+    int     (*platform_wakeup)(void);
+    void    (*exit_platform_hw)(void);
+};
+#endif
+
+#if defined (CONFIG_GPIOEXP_AW9523B)
+struct gpio_exp_platform_data {
+    int     (*init_platform_hw)(void);
+    void    (*exit_platform_hw)(void);
+};
+#endif
+
 enum _periph_pll {
 	periph_pll_1485mhz = 148500000,
 	periph_pll_297mhz = 297000000,
diff --git a/include/linux/rk_screen.h b/include/linux/rk_screen.h
index 8ad67254a786..a04c826899d5 100644
--- a/include/linux/rk_screen.h
+++ b/include/linux/rk_screen.h
@@ -48,6 +48,8 @@ struct rk29lcd_info {
 	u32	reset_pin;
 	int (*io_init)(void);
 	int (*io_deinit)(void);
+	int (*io_enable)(void);
+	int (*io_disable)(void);
 };
 
 
-- 
2.35.3

